server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/materials/index.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MithrilViewComponent} from "jsx/mithril-component";
+import _ from "lodash";
+import m from "mithril";
+import {Material, Materials} from "models/new_pipeline_configs/materials";
+import {Secondary} from "views/components/buttons";
+import {CollapsiblePanel} from "views/components/collapsible_panel";
+import {IconGroup} from "views/components/icons";
+import {Delete} from "views/components/icons";
+import {MaterialOperations} from "../../create_pipeline_page";
+import styles from "../index.scss";
+import materialStyles from "./index.scss";
+import {AddMaterialModal} from "./modals";
+
+interface Attrs {
+  materials: Materials;
+  materialOperations: MaterialOperations;
+}
+
+export class MaterialsWidget extends MithrilViewComponent<Attrs> {
+  view(vnode: m.Vnode<Attrs>) {
+    let materials = [];
+    let expanded  = true;
+    if (_.isEmpty(vnode.attrs) || _.isEmpty(vnode.attrs.materials) || vnode.attrs.materials.length === 0) {
```

Comment:
Don't think we need all these checks. `vnode.attrs` and `vnode.attrs.materials`  will always be defined due to the type safety of `Attrs`